compiler/rustc_trait_selection/src/traits/wf.rs

```diff
@@ -434,15 +434,19 @@ impl<'a, 'tcx> WfPredicates<'a, 'tcx> {
                             let obligations = self.nominal_obligations(def.did, substs);
                             self.out.extend(obligations);
 
-                            let predicate = ty::PredicateAtom::ConstEvaluatable(def, substs)
-                                .to_predicate(self.tcx());
-                            let cause = self.cause(traits::MiscObligation);
-                            self.out.push(traits::Obligation::with_depth(
-                                cause,
-                                self.recursion_depth,
-                                self.param_env,
-                                predicate,
-                            ));
+                            // FIXME(const_generics): Do we actually end up having to deal with
+                            // this bound later or is this unsound?
+                            if !substs.has_escaping_bound_vars() {
```

Comment:
cc @eddyb 

Can we somehow circumvent this? :thinking:

I don't think we can just replace the bound vars with placeholders here either.

I think that places which actually try to evaluate the constant do still have to satisfy the `ConstEvaluatable` predicate but it does allow for 
```rust
// check-pass
#![feature(const_generics)]
#![allow(incomplete_features)]

trait Baz<const N: usize> {}

fn test<T>() {
    // FIXME(const_generics): This should error.
    let _a: Box<dyn for<'a> Baz<{
        let _: &'a ();
        std::mem::size_of::<T>()
    }>>;
}

fn main() {
    test::<u32>();
}
```
which is less than ideal